src/backend/luna.rs

```diff
@@ -0,0 +1,100 @@
+use rusb::{
+    GlobalContext, DeviceHandle,
+};
+use std::collections::VecDeque;
+use std::thread;
+use std::sync::mpsc::{channel, Receiver};
+use std::time::Duration;
+
+const VID: u16 = 0x1d50;
+const PID: u16 = 0x615b;
+
+const ENDPOINT: u8 = 0x81;
+
+const READ_LEN: usize = 0x4000;
+
+pub struct LunaDevice {
+    handle: Option<DeviceHandle<GlobalContext>>,
+    worker: Option<thread::JoinHandle<DeviceHandle<GlobalContext>>>,
+    receiver: Option<Receiver<Vec<u8>>>,
+}
+
+impl LunaDevice {
+    pub fn open() -> Self {
+        let handle = rusb::open_device_with_vid_pid(VID, PID).unwrap();
+        LunaDevice {
+            handle: Some(handle),
+            worker: None,
+            receiver: None,
+        }
+    }
+
+    pub fn start(&mut self) {
+        let mut handle = self.handle.take().unwrap();
+        handle.claim_interface(0).expect("Failed to claim interface");
+        let (tx, rx) = channel();
+        self.receiver = Some(rx);
+        self.worker = Some(thread::spawn(move || {
+            let mut buffer = [0u8; READ_LEN];
+            let mut packet_queue = PacketQueue::new();
+            LunaDevice::enable_capture(&handle);
+            loop {
+                let result = handle.read_bulk(ENDPOINT, &mut buffer, Duration::from_secs(1));
+                if let Ok(count) = result {
+                    packet_queue.extend(&buffer[..count]);
+                    while let Some(packet) = packet_queue.next() {
+                        tx.send(packet).expect("Failed to send packet across channel");
+                    }
+                }
+            }
```

Comment:
This should do some amount of error handling. As it stands you can unplug the analyzer and it will just peg the CPU spinning in this loop. We should continue when `result` is `Err(Timeout)`, but any other error ought to cause this loop to exit.